server/mqtt_test.go

```diff
@@ -1699,39 +1699,6 @@ func TestMQTTDontSetPinger(t *testing.T) {
 	testMQTTPublish(t, mc, r, 0, false, false, "foo", 0, []byte("msg"))
 }
```

Comment:
Need to change test: `TestMQTTSubAck` to have `expected` slice change the second `1` to a `2` now since QoS2 is supported.

Also change `TestMQTTSubDups` so that we don't test the downgraded from QoS2 to 1 in this test:
```
-       // Ask for QoS 2 but server will downgrade to 1
-       testMQTTSub(t, 1, mc, r, []*mqttFilter{{filter: "bar", qos: 2}}, []byte{1})
+       // Ask for QoS 1
+       testMQTTSub(t, 1, mc, r, []*mqttFilter{{filter: "bar", qos: 1}}, []byte{1})
```